### PR TITLE
fix: pr changes master 

### PR DESCRIPTION
**Pull request by nrajanee**
## Description

fix: pr changes master. 
Splitting SearcherOperation and creating new TrialOperation. Return errors in fromProto methods. Remove CustomSearchMethod interface.

### CODE REVIEW

File: proto/src/determined/experiment/v1/searcher.proto [hunk 1]
```diff
@@ -165,12 +165,12 @@ message ShutdownOperation {
...
 }
 
-// SearcherOperation is an operation issued by the searcher.
+// SearcherOperation is an operation issued by the custom searcher.
 message SearcherOperation {
   // A searcher operation is one of the following operations.
   oneof union {
-    // ValidateAfter is issued to tell a trial to train some then validate.
-    ValidateAfterOperation validate_after = 1;
+    // TrialOperation is issued for the current trial. .
```
Comment (dzhu) [inline proto/src/determined/experiment/v1/searcher.proto:172]:
```suggestion
    // TrialOperation is issued to tell an existing trial to do something.
```

File: proto/src/determined/experiment/v1/searcher.proto [hunk 2]
```diff
@@ -165,12 +165,12 @@ message ShutdownOperation {
...
+    TrialOperation trial_op = 1;
```
Comment (dzhu) [inline proto/src/determined/experiment/v1/searcher.proto:173]: I'd prefer to have the name `trial_operation` so we don't have this mismatch between the type name and the field name.

File: proto/src/determined/experiment/v1/searcher.proto [hunk 3]
```diff
@@ -183,6 +183,17 @@ message SearcherOperation {
   }
 }
 
+// TrialOperation is issued for the current trial.
```
Comment (dzhu) [inline proto/src/determined/experiment/v1/searcher.proto:186]:
```suggestion
// TrialOperation is any operation that a trial can perform while it is active.
```

File: proto/src/determined/experiment/v1/searcher.proto [hunk 4]
```diff
@@ -183,6 +183,17 @@ message SearcherOperation {
...
+message TrialOperation {
+  // A trial operation is one of the following operations.
+  oneof union {
+    // ValidateAfter means a trial is currently training and is issued to tell
+    // the
+    //  trial to train some then validate.
```
Comment (dzhu) [inline proto/src/determined/experiment/v1/searcher.proto:192]:
```suggestion
    // ValidateAfter means a trial is currently training and will later
    // validate.
```

**Review by dzhu — APPROVED** [review]

File: proto/src/determined/experiment/v1/searcher.proto (hunk 1 above)
Comment (mpkouznetsov) [inline proto/src/determined/experiment/v1/searcher.proto:172]: I don't think this is quite what @dzhu had in mind.
Before Custom Searcher there already was something called "SearcherOperation". That something should become "TrialSearcherOperation" or "TrialOperation" if you prefer. It used to contain "ValidateAfterOperation", just like `TrialOperation` that you defined below.
However, it is _not_ that `TrialOperation` that we want to include into the new `SearcherOperation` but rather either `ValidateAfrterOperation` directly or something equivalent.

**Review by mpkouznetsov — APPROVED** [review]
makes sense

File: proto/src/determined/experiment/v1/searcher.proto (hunk 2 above)
Comment (nrajanee) [reply proto/src/determined/experiment/v1/searcher.proto]: yup sorry missed this.

**Pull request merged**